File: apps/desktop/src-tauri/src/models/ai_models.rs
#[derive(serde::Deserialize)]
pub struct AiModel {
    pub name: Option<String>,
    pub id: String,
    pub token_limit: u32,
    pub vendor: String,
}

#[derive(serde::Deserialize)]
pub struct ModelParams {
    pub temperature: Option<f64>,
    pub top_p: Option<f64>,
    pub top_k: Option<u32>,
    pub repeat_penalty: Option<f64>,
    pub presence_penalty: Option<f64>,
    pub stop: Option<Vec<String>>,
    pub max_tokens: Option<u32>,
    pub seed: Option<u32>,
}

File: apps/web/types/ai-models.ts
export type AiModel = {
  id: string;
  name: string;
  description?: string;
  strengths?: string;
  maxLength: number; // maximum length of a message
  tokenLimit: number;
  requestLimit: number;
  vendor: 'OpenAI' | 'Azure' | 'Anthropic' | 'Google' | 'Ollama';
  type: 'text';
} | {
  id: string;
  name: string;
  description?: string;
  strengths?: string;
  vendor: 'OpenAI' | 'Azure';
  type: 'image';
}

export const PossibleAiModelVendors = ['OpenAI', 'Anthropic', 'Google', 'Ollama'];

export type OpenAiClientOptions = {
  vendor: 'openai',
  apiUrl: string;
  organisation: string;
} | {
  modelId?: string;
  vendor: 'azure',
  apiUrl: string;
  apiVersion: string;
}

export interface GetAvailableAIModelResponse {
  error?: any;
  data: any[];
}

export interface ModelParams {
  temperature?: number;
  top_p?: number;
  top_k?: number;
  repeat_penalty?: number;
  presence_penalty?: number;
  stop?: string[];
  max_tokens?: number;
  seed?: number;
  n?: number | null;
  quality?: 'standard' | 'hd';
  response_format?: 'url' | 'b64_json' | null;
  size?: '256x256' | '512x512' | '1024x1024' | '1792x1024' | '1024x1792' | null;
  style?: 'vivid' | 'natural' | null;
}
